lib/iris/experimental/ugrid.py

```diff
@@ -737,358 +773,913 @@ def equal(self, other, lenient=None):
+          managers
+
+        - validate both managers contents e.g., shape? more...?
+
+    """
+
+    # TBD: for volume and/or z-axis support include axis "z" and/or dimension "3"
+    AXES = ("x", "y")
+    TOPOLOGY_DIMENSIONS = (1, 2)
+
+    def __init__(
+        self,
+        topology_dimension,
+        node_coords_and_axes,
+        standard_name=None,
+        long_name=None,
+        var_name=None,
+        units=None,
+        attributes=None,
+        edge_coords_and_axes=None,
+        face_coords_and_axes=None,
+        # connectivities=None,
+        node_dimension=None,
+        edge_dimension=None,
+        face_dimension=None,
+    ):
+        # TODO: support volumes.
+        # TODO: support (coord, "z")
+
+        self._metadata_manager = metadata_manager_factory(MeshMetadata)
+
+        # topology_dimension is read-only, so assign directly to the metadata manager
+        if topology_dimension not in self.TOPOLOGY_DIMENSIONS:
+            emsg = f"Expected 'topology_dimension' in range {self.TOPOLOGY_DIMENSIONS!r}, got {topology_dimension!r}."
+            raise ValueError(emsg)
+        self._metadata_manager.topology_dimension = topology_dimension
+
+        # TBD: these are strings, if None is provided then assign the default string.
+        self.node_dimension = node_dimension
+        self.edge_dimension = edge_dimension
+        self.face_dimension = face_dimension
+
+        # assign the metadata to the metadata manager
+        self.standard_name = standard_name
+        self.long_name = long_name
+        self.var_name = var_name
+        self.units = units
+        self.attributes = attributes
+
+        # based on the topology_dimension, create the appropriate coordinate manager
+        def normalise(location, axis):
+            result = str(axis).lower()
+            if result not in self.AXES:
+                emsg = f"Invalid axis specified for {location} coordinate {coord.name()!r}, got {axis!r}."
+                raise ValueError(emsg)
+            return f"{location}_{axis}"
+
+        kwargs = {}
+        for coord, axis in node_coords_and_axes:
+            kwargs[normalise("node", axis)] = coord
+        if edge_coords_and_axes is not None:
+            for coord, axis in edge_coords_and_axes:
+                kwargs[normalise("edge", axis)] = coord
+        if face_coords_and_axes is not None:
+            for coord, axis in face_coords_and_axes:
+                kwargs[normalise("face", axis)] = coord
+
+        # check the UGRID minimum requirement for coordinates
+        if "node_x" not in kwargs:
+            emsg = (
+                "Require a node coordinate that is x-axis like to be provided."
+            )
+            raise ValueError(emsg)
+        if "node_y" not in kwargs:
+            emsg = (
+                "Require a node coordinate that is y-axis like to be provided."
+            )
+            raise ValueError(emsg)
```

Comment:
I wonder if this intelligence should be part of the `coord_manager` instead.

Worth noting that the equivalent `Connectivity` validation is **definitely** at the `connectivity_manager` level since the required cf-roles are dependent on the manager type.